Tasks/GitHubCommentV0/task.json

```diff
@@ -20,7 +20,7 @@
     "inputs": [
         {
             "name": "gitHubConnection",
-            "type": "connectedService:github:OAuth,PersonalAccessToken,InstallationToken",
+            "type": "connectedService:github:OAuth,PersonalAccessToken,Token",
```

Comment:
This was intentional. If you want to add support to `token` concatenate `Token` to the `,` separated list. you'll have to validate any task side changes if required as well.

Reply:
@ds-ms The task side code already supports the Token scheme.  It was pretty odd for the code to support it but the task.json configuration to not have it as a usable option.

`    if (!!githubEndpointObject) {
        tl.debug('Endpoint scheme: ' + githubEndpointObject.scheme);

        if (githubEndpointObject.scheme === 'PersonalAccessToken') {
            githubEndpointToken = githubEndpointObject.parameters.accessToken;
        } else if (githubEndpointObject.scheme === 'OAuth') {
            githubEndpointToken = githubEndpointObject.parameters.AccessToken;
        } else if (githubEndpointObject.scheme === 'Token') {
            githubEndpointToken = githubEndpointObject.parameters.AccessToken;
        } else if (githubEndpointObject.scheme) {
            throw new Error(tl.loc('InvalidEndpointAuthScheme', githubEndpointObject.scheme));
        }
    }`

Reply:
@ds-ms Is there a reason for `InstallationToken` to be in the list? It looks like it only exists in such task descriptions in the repo, but it's probably a legacy.

Reply:
@ds-ms It's been some time since we started the conversation above. The change was tested and works as expected.

Reply:
@kohtenko sorry about the delay, I was off last week. 

`InstallationToken` isn't legacy per se. When you authorize `Azure Pipelines` GH app with your org, internally an InstallationToken is stored which is used for cloning your private GitHub repositories. This is an internal implementation, although as a user you can't use it your pipelines directly. 

This `InstallationToken` is required for our `Review App` scenario to work. When you setup a review app pipeline, we auto-inject the `GitHubCommentV0` task with the `InstallationToken` type service endpoint to post comments on the users PRs as `azure pipelines (bot)`

Like this: 
![image](https://user-images.githubusercontent.com/25762922/83840455-3cd90580-a71c-11ea-8d81-44178a96d22a.png)

Although when it reaches the task code it falls under the `Token` implementation. Given that `InstallationToken` is a privileged service connection we add additional checks on our server side to allow only authorized actions to use it.

If you remove `InstallationToken` the injected task's run would fail like this:

![image](https://user-images.githubusercontent.com/25762922/83840676-d30d2b80-a71c-11ea-8291-e08e1a32d220.png)


So in order to have our other features to work as well, I asked you to append to the current list. :smiley:

Reply:
I agree with @bdetchison I kinda missed adding `Token` to the supported list.

Reply:
@ds-ms this was a super helpful explanation.  I will add this back.